DSCResources/MSFT_xWindowsOptionalFeature/MSFT_xWindowsOptionalFeature.psm1

```diff
@@ -155,11 +155,11 @@ function Set-TargetResource
     {
         if ($NoWindowsUpdateCheck)
         {
-            $feature = Dism\Enable-WindowsOptionalFeature -FeatureName $Name -Online -LogLevel $DismLogLevel @PSBoundParameters -LimitAccess
+            $feature = Dism\Enable-WindowsOptionalFeature -FeatureName $Name -Online -LogLevel $DismLogLevel @PSBoundParameters -LimitAccess -NoRestart
         }
         else
         {
-            $feature = Dism\Enable-WindowsOptionalFeature -FeatureName $Name -Online -LogLevel $DismLogLevel @PSBoundParameters
+            $feature = Dism\Enable-WindowsOptionalFeature -FeatureName $Name -Online -LogLevel $DismLogLevel @PSBoundParameters -NoRestart
```

Comment:
@iainbrighton  the cmdlet return an object with the property RestartNeeded.  If it is true, the resource should set $global:DSCMachineStatus = 1, so that LCM will reboot the machine.  See [xPendingReboot](https://github.com/PowerShell/xPendingReboot/blob/dev/DSCResources/MSFT_xPendingReboot/MSFT_xPendingReboot.psm1#L51) 

## Example of a feature object 

```PowerShell
> $r =Enable-WindowsOptionalFeature -FeatureName telnetclient -Online -NoRestart

> $r


Path          :
Online        : True
RestartNeeded : False
```